test/TPP/passes/mlp-fusion.mlir

```diff
@@ -0,0 +1,31 @@
+// RUN: tpp-opt %s -map-linalg-to-tpp -tile-consumer-and-fuse-producers="tile-sizes=32,32" -canonicalize -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map"  -canonicalize -drop-equivalent-buffer-results -finalizing-bufferize -convert-linalg-to-tpp | FileCheck %s
+
+#map0 = affine_map<(d0, d1) -> (d1)>
+#map1 = affine_map<(d0, d1) -> (d0, d1)>
+#map2 = affine_map<(d0, d1, d2) -> (d0, d2)>
+#map3 = affine_map<(d0, d1, d2) -> (d2, d1)>
+#map4 = affine_map<(d0, d1, d2) -> (d0, d1)>
+  
+func.func @main(%arg0: tensor<128x256xf32>, %arg1: tensor<256x512xf32>,
+  %arg2: tensor<512xf32>,  %output: tensor<128x512xf32>) -> tensor<128x512xf32> {
+  // CHECK: tpp.identity
+  %1 = linalg.generic {indexing_maps = [#map0, #map1], iterator_types = ["parallel", "parallel"]} ins(%arg2 : tensor<512xf32>) outs(%output : tensor<128x512xf32>) {
+    ^bb0(%arg9: f32, %arg10: f32):
+      linalg.yield %arg9 : f32
+  } -> tensor<128x512xf32>
+  // CHECK-NEXT: tpp.matmul
+  %2 = linalg.generic {indexing_maps = [#map2, #map3, #map4], iterator_types = ["parallel", "parallel", "reduction"]} ins(%arg0, %arg1 : tensor<128x256xf32>, tensor<256x512xf32>) outs(%1 : tensor<128x512xf32>) attrs =  {iterator_ranges = [128, 512, 256]} {
+    ^bb0(%arg9: f32, %arg10: f32, %arg11: f32):
+      %16 = arith.mulf %arg9, %arg10 : f32
+      %17 = arith.addf %arg11, %16 : f32
+      linalg.yield %17 : f32
+  } -> tensor<128x512xf32>
+  // CHECK-NEXT: tpp.relu
```

Comment:
IIUC, this is just fusing `id + matmul + relu` inside the loop, not fusing TPP ops, right?

We should have a second RUN line that goes all the way down fusing TPP ops too, later on.

Reply:
yes, we don't have fusion on tpps call. We just fuse at linalg level and maps to tpp.